src/ast/semantic_analyser.cpp

```diff
@@ -94,6 +94,15 @@ void SemanticAnalyser::visit(Builtin &builtin)
         err_ << "BPF_FUNC_get_current_cgroup_id is not available for your kernel version" << std::endl;
       #endif
     }
+    /*
+     * If we are using curtak in field access, retype it
+     * to its original type: struct task_truct.
+     */
+    if (builtin.ident == "curtask" && builtin.is_field_access) {
```

Comment:
Would people ever not want `curtask` defined as a struct like this? I'd say always treat it as a `struct task_struct *` and get rid of `is_field_access`

Reply:
ok, will check